modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx

```diff
@@ -309,23 +333,38 @@ export default function ActionBuilder({
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, []);
 
-	const hasEmptyValues = values.parameters?.predefinedValues?.some((item) =>
-		invalidateRequired(item.value)
-	);
+	useEffect(() => {
+		let error = '';
+		if (hasEmptyValues) {
+			error = Liferay.Language.get(
+				'required-fields-must-have-predefined-values'
+			);
+		}
+		else if (hasSyntaxError) {
+			error = Liferay.Language.get('syntax-error');
+		}
 
-	const predefinedValuesAlertMessage =
-		!hasEmptyValues && Object.keys(errors).length > 1
-			? Liferay.Language.get('syntax-error')
-			: Liferay.Language.get(
-					'required-fields-must-have-predefined-values'
-			  );
+		if (error) {
+			setPredefinedValuesAlertMessage(error);
+			setErrorAlert(true);
+		}
+		else {
+			setErrorAlert(false);
+		}
+	}, [
+		hasEmptyValues,
+		hasSyntaxError,
+		setErrorAlert,
+		setPredefinedValuesAlertMessage,
+	]);
```

Comment:
It no seems the case for a `useEffect()`, you probably may replace this for a `useMemo()`